master/internal/api_notebook.go

```diff
@@ -162,13 +173,55 @@ func (a *apiServer) SetNotebookPriority(
 	return resp, a.ask(notebooksAddr.Child(req.NotebookId), req, &resp)
 }
 
-func (a *apiServer) LaunchNotebook(
-	ctx context.Context, req *apiv1.LaunchNotebookRequest,
-) (*apiv1.LaunchNotebookResponse, error) {
+// isNTSCPermittedToLaunch checks authorization to launch in a given
+// workspace.
+func (a *apiServer) isNTSCPermittedToLaunch(
+	ctx context.Context, spec *tasks.GenericCommandSpec,
+) error {
+	workspaceID := spec.Metadata.WorkspaceID
+	if workspaceID == 0 {
+		panic("workspace ID must be set")
+	}
+
 	user, _, err := grpcutil.GetUser(ctx)
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get the user: %s", err)
+		return status.Errorf(codes.Internal, "failed to get the user: %s", err)
+	}
+
+	w := &workspacev1.Workspace{}
+	notFoundErr := status.Errorf(codes.NotFound, "workspace (%d) not found", workspaceID)
+	if err := a.m.db.QueryProto(
+		"get_workspace", w, workspaceID, user.ID,
+	); errors.Is(err, db.ErrNotFound) {
+		return notFoundErr
+	} else if err != nil {
+		return errors.Wrapf(err, "error fetching workspace (%d) from database", workspaceID)
+	}
+	if w.Archived {
+		// CHAT: we might want to use a.GetWorkspaceByID here instead.
+		// Do we allow you to launch into a workspace that you don't have view access to?
+		return notFoundErr
+	}
+
+	if spec.TaskType == model.TaskTypeTensorboard {
+		if ok, err := command.AuthZProvider.Get().CanGetTensorboard(
+			ctx, *user, workspaceID); err != nil || !ok {
+			return err
```

Comment:
on the topic of keeping these consistent we might want a map and filter here. this would become clear once we add tests for tensorboards here 
https://github.com/hamidzr/determined/blob/2e16fc41f2a98a7afb8a124f1ad92b8d43c169e5/master/internal/api_ntsc_intg_test.go#L151